ansible/roles/backup/tasks/main.yml

```diff
@@ -0,0 +1,149 @@
+---
+# Stop services
+- name: Stop js-ceramic-4 node
+  ansible.builtin.systemd:
+    name: js-ceramic
+    state: stopped
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+
+- name: Stop go-ipfs Docker container
+  ansible.builtin.shell:
+    cmd: docker-compose down
+    chdir: /opt/go-ipfs
+  when: inventory_hostname == 'gitcoin-go-ipfs-2.3boxlabs.com'
+  register: docker_compose_result
+
+# Ceramic backup
+- name: Get current timestamp
+  ansible.builtin.set_fact:
+    timestamp: "{{ ansible_date_time.iso8601_basic_short }}"
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+
+- name: Backup Ceramic daemon config file with timestamp
+  ansible.builtin.copy:
+    src: "{{ data_store_mount_path }}/daemon.config.json"
+    dest: "~/ceramic_config_backup_{{ timestamp }}"
+    remote_src: yes
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+
+- name: Get latest common snapshot between gitcoin-js-ceramic-3 and gitcoin-js-ceramic-4
+  ansible.builtin.shell:
+    cmd: |
+      local_snapshots=$(zfs list -H -t snapshot -o name ceramiconepool/data-store)
+      remote_snapshots=$(gcloud compute ssh gitcoin-js-ceramic-3 --zone "us-central1-a" --command="zfs list -H -t snapshot -o name ceramiconepool/data-store")
+      echo "$local_snapshots" | grep -Fx "$remote_snapshots" | tail -n 1
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+  register: ceramic_common_snapshot
+  failed_when: ceramic_common_snapshot.rc != 0 or ceramic_common_snapshot.stdout == ""
+  changed_when: false
+  become: yes
+
+- name: Create new ZFS snapshot on gitcoin-js-ceramic-3
+  ansible.builtin.shell:
+    cmd: |
+      gcloud compute ssh gitcoin-js-ceramic-3 --zone "us-central1-a" --command="
+        snapshot_name=ceramiconepool/data-store@migration_$(date +%Y%m%d_%H%M%S)
+        zfs snapshot \$snapshot_name
+        echo \$snapshot_name
+      "
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
```

Comment:
What am I missing, why wouldn't we just do this on gitcoin-js-ceramic-3?

Reply:
GC4 is the backup node and will _pull_ the latest snapshot from GC3. Taking this role out for now.